db/system_keyspace.cc

```diff
@@ -1503,6 +1503,27 @@ future<> system_keyspace::cache_truncation_record() {
     });
 }
 
+future<> system_keyspace::drop_truncation_rp_records() {
+    sstring req = format("SELECT table_uuid, shard, segment_id from system.{}", TRUNCATED);
+    auto rs = co_await execute_cql(req);
+
+    bool any = false;
+    co_await parallel_for_each(rs->begin(), rs->end(), [&] (const cql3::untyped_result_set_row& row) -> future<> {
```

Comment:
There is coroutine::parallel_for_each(). Also, please check if the inner coroutine is susceptible to the lambda coroutine fiasco (I think with coroutine::parallel_for_each it is not).

Reply:
Should not be. The only wait-point (execute_cql) uses only framelocal var (req), not references held by lambda.

Reply:
[&] itself captures stuff. It will get copied into parallel_for_each's stack frame.

Reply:
anyway, it's better not to rely on random features of the inner lambda, and instead us coroutine::parallel_for_each()

Reply:
& captures refs into the lambda object, which can die on a co_await in the lambda frame, but all refs inherited are used before co_await (line 1519), so should not be any risk of using stale refs.